pkg/upload/dsym.go

```diff
@@ -0,0 +1,128 @@
+package upload
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/bugsnag/bugsnag-cli/pkg/ios"
+	"github.com/bugsnag/bugsnag-cli/pkg/log"
+	"github.com/bugsnag/bugsnag-cli/pkg/server"
+	"github.com/bugsnag/bugsnag-cli/pkg/utils"
+)
+
+type Dsym struct {
+	VersionName string      `help:"The version of the application."`
+	Scheme      string      `help:"The name of the scheme to use when building the application."`
+	Dev         bool        `help:"Indicates whether the application is a debug or release build"`
+	Plist       string      `help:"Path to the Info.plist file" type:"path"`
+	ProjectRoot string      `help:"path to remove from the beginning of the filenames in the mapping file" type:"path"`
+	Path        utils.Paths `arg:"" name:"path" help:"Path to directory or file to upload" type:"path" default:"."`
+}
+
+func ProcessDsym(
+	apiKey string,
+	versionName string,
+	scheme string,
+	dev bool,
+	plistPath string,
+	projectRoot string,
+	paths []string,
+	endpoint string,
+	timeout int,
+	retries int,
+	overwrite bool,
+	dryRun bool,
+) error {
+
+	var buildSettings *ios.XcodeBuildSettings
+
+	for _, path := range paths {
+		uploadInfo, err := ios.ProcessPathValue(path, projectRoot)
+		if err != nil {
+			return err
+		}
+
+		// If projectRoot is not set explicitly, use uploadInfo to set it
+		if projectRoot == "" {
+			projectRoot = uploadInfo.ProjectRoot
+		}
+
+		// If scheme is set explicitly, check if it exists
+		if scheme != "" {
+			_, err = ios.IsSchemeInPath(path, scheme, projectRoot)
+			if err != nil {
+				return err
+			}
+		} else {
+			// If the scheme is not set explicitly, try to find it
+			scheme, err = ios.GetDefaultScheme(path, projectRoot)
```

Comment:
Do we only need `scheme` if the dsym path is not given? If so, we shouldn't error if we can't find a default scheme as that won't be needed.

Reply:
Ah - Yeah we're only dealing with the scheme if the dsym path is not given. Maybe I can just gate this whole block of code behind a condition e.g. `if uploadInfo.DsymPath == ""`